server/athenian/api/controllers/miners/github/pull_request.py

```diff
@@ -35,33 +37,55 @@ def __init__(self, prs: pd.DataFrame, reviews: pd.DataFrame, review_comments: pd
+        async with db.connection() as conn:
+            prs = await read_sql_query(select([PullRequest]).where(sql.and_(*filters)),
+                                       conn, PullRequest)
+            numbers = prs[PullRequest.number.key] if len(prs) > 0 else set()
+            reviews = await read_sql_query(select([PullRequestReview]).where(
+                sql.and_(PullRequestReview.pull_request_number.in_(numbers),
+                         PullRequestReview.repository_fullname.in_(repositories))),
+                conn, PullRequestReview)
+            review_comments = await read_sql_query(select([PullRequestComment]).where(
+                sql.and_(PullRequestComment.pull_request_number.in_(numbers),
+                         PullRequestComment.repository_fullname.in_(repositories))),
+                conn, PullRequestComment)
+            commits = await read_sql_query(select([PullRequestCommit]).where(
+                sql.and_(PullRequestCommit.pull_request_number.in_(numbers),
+                         PullRequestCommit.repository_fullname.in_(repositories))),
+                conn, PullRequestCommit)
+        return cls(prs, reviews, review_comments, commits, time_to)
+
+    def __iter__(self) -> Generator[Tuple[pd.Series, pd.DataFrame, pd.DataFrame, pd.DataFrame],
+                                    None, None]:
         """Iterate over the information collected for individual pull requests."""
+        r_repo_key = PullRequest.repository_fullname.key
+        r_pull_request_number_key = PullRequestReview.pull_request_number.key
+        ce_pull_request_number_key = PullRequestComment.pull_request_number.key
+        ce_repo_key = PullRequestComment.repository_fullname.key
+        ci_pull_request_number_key = PullRequestCommit.pull_request_number.key
+        ci_repo_key = PullRequestCommit.repository_fullname.key
+        number_key = PullRequest.number.key
+        repo_key = PullRequest.repository_fullname.key
         for _, pr in self._prs.iterrows():
-            reviews = self._reviews[self._reviews["pull_request_number"] == pr["number"]]
+            pr_number = pr[number_key]
+            pr_repo = pr[repo_key]
+            reviews = self._reviews[
+                (self._reviews[r_pull_request_number_key] == pr_number)
+                & (self._reviews[r_repo_key] == pr_repo)]  # noqa: W503
             review_comments = self._review_comments[
-                self._review_comments["pull_request_number"] == pr["number"]]
-            yield pr, reviews, review_comments
+                (self._review_comments[ce_pull_request_number_key] == pr_number)
+                & (self._review_comments[ce_repo_key] == pr_repo)]  # noqa: W503
+            commits = self._commits[
+                (self._commits[ci_pull_request_number_key] == pr_number)
+                & (self._commits[ci_repo_key] == pr_repo)]  # noqa: W503
+            yield pr, reviews, review_comments, commits
```

Comment:
I think that something as follows would be more readable (btw I didn't test it):
```
    def __iter__(self) -> Generator[Tuple[pd.Series, pd.DataFrame, pd.DataFrame, pd.DataFrame],
                                    None, None]:
        keys = {
            'reviews': {
                'pr': PullRequestReview.pull_request_number.key,
                'repo': PullRequestReview.repository_fullname.key,
            },
            'review_comments': {
                'pr': PullRequestComment.pull_request_number.key,
                'repo': PullRequestComment.repository_fullname.key,
            },
            'commits': {
                'pr': PullRequestCommit.pull_request_number.key,
                'repo': PullRequestCommit.repository_fullname.key 
            }
        }

        def _get_filtered(what, pr_number, pr_repo):
            df = getattr(self, f'_{what}')
            pr_key, repo_key = keys[what]['pr'], keys[what]['key']
            return df[(df[pr_key] == pr_number) & (df[repo_key] == pr_repo)]
        
        number_key = PullRequest.number.key
        repo_key = PullRequest.repository_fullname.key
        for _, pr in self._prs.iterrows():
            pr_number, pr_repo = pr[number_key], pr[repo_key]

            reviews = _get_filtered('reviews', pr_number, pr_repo)
            reviews_comments = _get_filtered('review_comments', pr_number, pr_repo)
            commits = _get_filtered('commits', pr_number, pr_repo)

            yield pr, reviews, review_comments, commits
```